scripts/page_browse.js

```diff
@@ -0,0 +1,247 @@
+};
+
+var projectSelectButton = function() {
+    return $("<input>", {type: 'submit', value: dpData.strings.selectProject});
+};
+
+var hiddenMode = function() {
+    return $("<input>", {type: 'hidden', name: 'mode', value: mode});
+};
+
+var topLine = function() {
+    let resetButton = $("<input>", {type: 'submit', value: dpData.strings.reset});
+    let hiddenProjectInput = $("<input>", {type: 'hidden', name: 'project', value: pageBrowserData.projectid});
+
+    resetButton.click(function () {
+        hiddenProjectInput.val("");
+    });
+    return $("<p>").append(pageBrowserData.heading, resetButton, hiddenProjectInput);
+};
+
+// Construct the controls to change pages
+var pageChanger = function () {
+    let pageSelector = document.createElement("select");
+    pageSelector.name = 'imagefile';
+    let prevButton = $("<input>", {type: 'submit', value: dpData.strings.previous});
+    let nextButton = $("<input>", {type: 'submit', value: dpData.strings.next});
+
+    pageBrowserData.pages.forEach(function(page) {
+        let selected = pageBrowserData.currentPage === page;
+        pageSelector.add(new Option(page, page, selected, selected));
+    });
+
+    $(pageSelector).change(function() {
+        this.form.submit();
+    });
+
+    prevButton.prop("disabled", pageSelector.selectedIndex === 0);
+    nextButton.prop("disabled", pageSelector.selectedIndex === (pageSelector.length - 1));
+
+    prevButton.click(function () {
+        pageSelector.selectedIndex -= 1;
+    });
+
+    nextButton.click(function () {
+        pageSelector.selectedIndex += 1;
+    });
+    return $("<span>", {class: "nowrap"}).append(dpData.strings.page + " ", pageSelector, prevButton, nextButton);
+};
+
+// Construct the buttons for horizontal/vertical split
+// and return a splitter variable.
+var viewSplitter = function(container) {
+    const textImageSplitID = "text_image_split";
+    const splitPercentID = "split_percent";
+    let splitter = splitControl();
+
+    // stored value is "horizontal" or "vertical"
+    let splitDirString = localStorage.getItem(textImageSplitID);
+    if(!splitDirString) {
+        splitDirString = "horizontal";
+    }
+    // splitDirection is a value defined in splitter
+    let splitDirection = (splitDirString === "horizontal") ? splitter.DIRECTION.HORIZONTAL : splitter.DIRECTION.VERTICAL;
+
+    let splitPercent = localStorage.getItem(splitPercentID);
+    if(!splitPercent) {
+        splitPercent = 50;
+    }
+
+    let mainSplit = splitter.setup(container, {splitDirection: splitDirection, splitPercent: splitPercent});
+
+    // must set the image width, otherwise it will at first be small, then when image loads later elements can wrap and disappear
+    let vSwitchButton = $("<button>", {type: 'button', class: 'img-button', title: dpData.strings.switchVert}).append($("<img>", {src: dpData.buttonImages.imgVSplit, width: 27}));
```

Comment:
Ideally we wouldn't include the size of the images here, otherwise if we ever change them we have to remember to come back here and change it, and since it isn't colocated with the name of the image itself we'll likely never find it.

Reply:
Changed to a relaying out after images are loaded whic seems to work.